app/scripts/components/common/nav-wrapper.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import T from 'prop-types';
+import styled from 'styled-components';
+import PageHeader from './page-header';
+
+import PageLocalNav from '$components/common/page-local-nav';
+import useScrollDirection from '$utils/use-scroll-direction';
+
+const NavWrapper = styled.div`
+  position: sticky;
+  top: 0;
+  z-index: 1000;
+`;
+
+const AnimatedPageHeader = styled(PageHeader)`
+  transition: margin-top 0.32s ease-out;
+  margin-top: 0;
+  &.up {
+    margin-top: 0;
+  }
+  &.down {
+    margin-top: calc(
+      -2.5rem - (1rem * var(--base-space-multiplier, 1) * 0.75 * 2)
+    );
+  }
+`;
+
+function PageNavWrapper({ localNavProps }) {
+  const renderLocalNav = !!localNavProps;
+  const scrollDir = useScrollDirection();
+
+  return (
+    <NavWrapper>
+      <AnimatedPageHeader className={scrollDir} />
```

Comment:
I'd like to pick your brain as to why use classes instead of styled-component props. Do you feel this performs better?

Reply:
No your comment does make me wonder if it is more performative to use classname than dynamic props (possibly? classname will just direct styles for existing css compare to props changing styles dynamically?) In this case, I think using props might be better to respond to scroll direction's value so I changed it to use props.